accounts-db/src/append_vec.rs

```diff
@@ -728,9 +729,16 @@ impl AppendVec {
             }
             AppendVecFileBacking::File(file) => {
                 // 4096 was just picked to be a single page size
-                let mut buf = [0u8; PAGE_SIZE as usize];
-                let bytes_read = read_into_buffer(file, self.len(), offset, &mut buf).ok()?;
-                let valid_bytes = ValidSlice(&buf[..bytes_read]);
+                let mut buf = [MaybeUninit::<u8>::uninit(); PAGE_SIZE as usize];
+                // SAFETY: `read_into_buffer` will only write to uninitialized memory.
+                let bytes_read = read_into_buffer(file, self.len(), offset, unsafe {
+                    slice::from_raw_parts_mut(buf.as_mut_ptr() as *mut u8, buf.len())
+                })
+                .ok()?;
+                // SAFETY: we only read the initialized portion.
+                let valid_bytes = ValidSlice(unsafe {
+                    slice::from_raw_parts(buf.as_ptr() as *const u8, bytes_read)
+                });
```

Comment:
Way `ArrayVec` considered at all? The crate holds an array of `MaybeUninit` under the hood so essentially the same. But, using the type gives us the nice abstraction / avoids the unsafes in our code. If I'm not mistaken, I think the following snippet would be comparable:
```rust
let mut buffer = ArracyVec<u8, PAGE_SIZE>::new();
let bytes_read = read_into_buffer(file, self.len(), offset, &mut buffer).ok()?;
// Technically don't need read_into_buffer() to track bytes read anymore
debug_assert!(bytes_read, buffer.len());
let valid_bytes = ValidSlice(buffer.as_slice());
```
https://docs.rs/arrayvec/latest/arrayvec/struct.ArrayVec.html

Reply:
wouldn't work, read_into_buffer calls file.read_at() which wants `&mut [u8]` and arrayvec.as_mut_slice() would return the empty slice